iml-manager-cli/src/server.rs

```diff
@@ -349,7 +365,8 @@ pub async fn server_cli(command: ServerCommand) -> Result<(), ImlManagerCliError
             let objects: Vec<_> = objects
                 .into_iter()
                 .filter_map(|x| x.host_profiles)
-                .filter(|x| is_profile_valid(&x, &profile.name))
+                .filter(|x| is_profile_valid(&x, &profile.name) ||
```

Comment:
I think we'll need to split this into two phases.

The first phase is simply checking that the profile selected by the user exists in the objects being iterated over. If the profile does not exist in these objects, then we want to return early; there is no point in overriding / prompting if we can not deploy the profile we want.

The second phase will check if all the profiles checks have passed. If they have not passed, this is where we want to use the prompt / override logic.

For the first phase, if we use `find`, we can pick out the single profile we're interested in. 

Then we can look at that profile and check it's items to see if they all pass or not. If we continue forward we can then convert it into a `HostProfileConfig` and continue onwards.

Reply:
1. If no profile then always filter out object, right?
2. Do you mean use ```iter().find()``` instead of ```get()```? Maybe ```contains_key()```?